front_end/src/components/random_button.tsx

```diff
@@ -0,0 +1,41 @@
+"use client";
+
+import { useRouter } from "next/navigation";
+import { FC, useState } from "react";
+
+import { fetchRandomPostId } from "@/app/(main)/questions/actions";
+import Button from "@/components/ui/button";
+
+import { Die } from "./icons/die";
+import "./icons/die.css";
+
+const RandomButton: FC = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const router = useRouter();
+
+  const handleRandomClick = async () => {
+    setIsLoading(true);
+    try {
+      const data = await fetchRandomPostId();
+      router.push(`/questions/${data.id}/${data.url_title}/`);
```

Comment:
We have different urls for questions from communities and notebooks, so check `getPostLink` function (probably we'll need to update endpoint to return all required post data for it).

Alternatively, you can narrow down Post response to questions only and not to overload BE response with extra params

Reply:
Oh, right. I'll restrict the endpoint to only Question type Posts.